packages/mail-client/src/mail-client.spec.ts

```diff
@@ -258,16 +239,69 @@ describe('mail client', () => {
       from: 'from1@example.com',
       to: 'to1@example.com'
     };
-    await expect(
-      sendMail(destination, mailOptions, { sdkOptions: { parallel: false } })
-    ).resolves.not.toThrow();
-    expect(spyCreateSocket).toBeCalledTimes(1);
-    expect(spyCreateTransport).toBeCalledTimes(1);
-    expect(spySendMail).toBeCalledTimes(1);
-    expect(spySendMail).toBeCalledWith(mailOptions);
-    expect(spyCloseTransport).toBeCalledTimes(1);
-    expect(spyEndSocket).toBeCalledTimes(1);
-    expect(spyDestroySocket).toBeCalledTimes(1);
+
+    it('[OnPrem] should create transport/socket, send mails and close the transport/socket', async () => {
+      const { mockSocket, createConnectionSpy } = mockSocketConnection();
+      const spyCreateTransport = jest
+        .spyOn(nodemailer, 'createTransport')
+        .mockReturnValue(mockTransport as any);
+      const spySendMail = jest
+        .spyOn(mockTransport, 'sendMail')
+        .mockImplementation(() => {
+          mockSocket.socket.on('data', () => {});
+        });
+      const spyCloseTransport = jest.spyOn(mockTransport, 'close');
+      const spyEndSocket = jest.spyOn(mockSocket.socket, 'end');
+      const spyDestroySocket = jest.spyOn(mockSocket.socket, 'destroy');
+
+      await expect(
+        sendMail(destination, mailOptions, { sdkOptions: { parallel: false } })
+      ).resolves.not.toThrow();
+      expect(createConnectionSpy).toHaveBeenCalledTimes(1);
+      expect(spyCreateTransport).toHaveBeenCalledTimes(1);
+      expect(spySendMail).toHaveBeenCalledTimes(1);
+      expect(spySendMail).toHaveBeenCalledWith(mailOptions);
+      expect(spyCloseTransport).toHaveBeenCalledTimes(1);
+      expect(spyEndSocket).toHaveBeenCalledTimes(1);
+      expect(spyDestroySocket).toHaveBeenCalledTimes(1);
+    });
+
+    it('[OnPrem] should resend greeting', async () => {
+      const { mockSocket } = mockSocketConnection();
+      jest
+        .spyOn(nodemailer, 'createTransport')
+        .mockReturnValue(mockTransport as any);
+      jest.spyOn(mockTransport, 'sendMail').mockImplementation(() => {
+        mockSocket.socket.on('data', () => {});
+      });
+
+      const req = sendMail(destination, mailOptions, {
+        sdkOptions: { parallel: false }
+      });
+
+      const emitsTwice = new Promise<void>(resolve => {
+        let dataEmitCount = 0;
+        mockSocket.socket.on('data', data => {
+          dataEmitCount += data.toString().includes('220') ? 1 : 0;
+          if (dataEmitCount === 2) {
```

Comment:
[q] Can you explain a bit why the on data event is emitted twice and write some comments here?

Reply:
I did it in the test now, let me know if this is what you meant.

Reply:
Exactly what I meant! Thanks for adding this.